src/WebJobs.Script/Description/FSharp/FSharpCompiler.cs

```diff
@@ -50,47 +50,99 @@ public ICompilation GetFunctionCompilation(FunctionMetadata functionMetadata)
+            FSharpErrorInfo[] errors = null;
+
+            //var exitCode = result.Item2;
+            FSharpOption<Assembly> assemblyOption = null;
+            try
             {
-                File.AppendAllLines(scriptFile, new string[] { "open " + import });
-            }
-            File.AppendAllText(scriptFile, File.ReadAllText(functionMetadata.ScriptFile));
-            var otherFlags = new List<string>();
+                // Write an adjusted version of the script file, prefixing some 'open' decarations
+                foreach (var import in script.Options.Imports)
+                {
+                    File.AppendAllLines(scriptFile, new string[] { "open " + import });
+                    
+                }
+                File.AppendAllLines(scriptFile, new string[] { "# 0 @\"" + functionMetadata.ScriptFile + "\"" });
+                var scriptSource = File.ReadAllText(functionMetadata.ScriptFile);
+                File.AppendAllText(scriptFile, scriptSource);
+                var otherFlags = new List<string>();
 
-            // For some reason CompileToDynamicAssembly wants "fsc.exe" as the first arg, it is ignored.
-            otherFlags.Add("fsc.exe");
+                // For some reason CompileToDynamicAssembly wants "fsc.exe" as the first arg, it is ignored.
+                otherFlags.Add("fsc.exe");
 
-            otherFlags.Add("--noframework");
+                // The --noframework option is used because we will shortly add references to mscorlib and FSharp.Core
+                // as dictated by the C# reference resolver, and F# doesn't like getting multiple references to those.
+                otherFlags.Add("--noframework");
 
-            foreach (var mdr in compilation.References)
-            {
-                if (!mdr.Display.Contains("Unresolved "))
+                // Add the references as reported by the C# reference resolver.
+                foreach (var mdr in compilation.References)
                 {
-                    otherFlags.Add("-r:" + mdr.Display);
+                    if (!mdr.Display.Contains("Unresolved "))
+                    {
+                        otherFlags.Add("-r:" + mdr.Display);
+                    }
                 }
-            }
 
-            otherFlags.Add("--optimize+");
-            if (debug)
-            {
-                otherFlags.Add("--debug+");
-                //otherFlags.Add("--tailcalls-");
-            }
+                // Above we have used the C# reference resolver to get the basic set of DLL references for the compilation.
+                //
+                // However F# has its own view on default options. For scripts these should include the
+                // following framework facade references.
+
+                otherFlags.Add("-r:System.Linq.dll"); // System.Linq.Expressions.Expression<T> 
```

Comment:
@fabiocav This explicit set of DLL references is unfortunate, but I think OK for now.  We will look at updating FCS some mechanism to give this list